flink-table/flink-sql-parser/src/test/java/org/apache/flink/sql/parser/FlinkDDLDataTypeTest.java

```diff
@@ -0,0 +1,441 @@
+			createTestItem("BINARY(33)", nullable(FIXTURE.binary33Type), "BINARY(33)"),
+			createTestItem("VARBINARY", nullable(FIXTURE.varbinaryType), "VARBINARY"),
+			createTestItem("VARBINARY(33)", nullable(FIXTURE.varbinary33Type),
+				"VARBINARY(33)"),
+			createTestItem("BYTES",
+				nullable(FIXTURE.createSqlType(SqlTypeName.VARBINARY, Integer.MAX_VALUE)),
+				"BYTES"),
+			createTestItem("DECIMAL", nullable(FIXTURE.decimalType), "DECIMAL"),
+			createTestItem("DEC", nullable(FIXTURE.decimalType), "DECIMAL"),
+			createTestItem("NUMERIC", nullable(FIXTURE.decimalType), "DECIMAL"),
+			createTestItem("DECIMAL(10)", nullable(FIXTURE.decimalP10S0Type), "DECIMAL(10)"),
+			createTestItem("DEC(10)", nullable(FIXTURE.decimalP10S0Type), "DECIMAL(10)"),
+			createTestItem("NUMERIC(10)", nullable(FIXTURE.decimalP10S0Type), "DECIMAL(10)"),
+			createTestItem("DECIMAL(10, 3)", nullable(FIXTURE.decimalP10S3Type),
+				"DECIMAL(10, 3)"),
+			createTestItem("DEC(10, 3)", nullable(FIXTURE.decimalP10S3Type),
+				"DECIMAL(10, 3)"),
+			createTestItem("NUMERIC(10, 3)", nullable(FIXTURE.decimalP10S3Type),
+				"DECIMAL(10, 3)"),
+			createTestItem("TINYINT", nullable(FIXTURE.tinyintType), "TINYINT"),
+			createTestItem("SMALLINT", nullable(FIXTURE.smallintType), "SMALLINT"),
+			createTestItem("INTEGER", nullable(FIXTURE.intType), "INTEGER"),
+			createTestItem("INT", nullable(FIXTURE.intType), "INTEGER"),
+			createTestItem("BIGINT", nullable(FIXTURE.bigintType), "BIGINT"),
+			createTestItem("FLOAT", nullable(FIXTURE.floatType), "FLOAT"),
+			createTestItem("DOUBLE", nullable(FIXTURE.doubleType), "DOUBLE"),
+			createTestItem("DOUBLE PRECISION", nullable(FIXTURE.doubleType), "DOUBLE"),
+			createTestItem("DATE", nullable(FIXTURE.dateType), "DATE"),
+			createTestItem("TIME", nullable(FIXTURE.timeType), "TIME"),
+			createTestItem("TIME WITHOUT TIME ZONE", nullable(FIXTURE.timeType), "TIME"),
+			createTestItem("TIME(3)", nullable(FIXTURE.time3Type), "TIME(3)"),
+			createTestItem("TIME(3) WITHOUT TIME ZONE", nullable(FIXTURE.time3Type),
+				"TIME(3)"),
+			createTestItem("TIMESTAMP", nullable(FIXTURE.timestampType), "TIMESTAMP"),
+			createTestItem("TIMESTAMP WITHOUT TIME ZONE", nullable(FIXTURE.timestampType),
+				"TIMESTAMP"),
+			createTestItem("TIMESTAMP(3)", nullable(FIXTURE.timestamp3Type), "TIMESTAMP(3)"),
+			createTestItem("TIMESTAMP(3) WITHOUT TIME ZONE",
+				nullable(FIXTURE.timestamp3Type), "TIMESTAMP(3)"),
+			createTestItem("TIMESTAMP WITH LOCAL TIME ZONE",
+				nullable(FIXTURE.timestampWithLocalTimeZoneType),
+				"TIMESTAMP WITH LOCAL TIME ZONE"),
+			createTestItem("TIMESTAMP(3) WITH LOCAL TIME ZONE",
+				nullable(FIXTURE.timestamp3WithLocalTimeZoneType),
+				"TIMESTAMP(3) WITH LOCAL TIME ZONE"),
+			createTestItem("ARRAY<TIMESTAMP(3) WITH LOCAL TIME ZONE>",
+				nullable(FIXTURE.createArrayType(nullable(FIXTURE.timestamp3WithLocalTimeZoneType))),
+				"ARRAY< TIMESTAMP(3) WITH LOCAL TIME ZONE >"),
```

Comment:
Tests for intervals are missing.

Reply:
There is one, because `INTERVAL` is not supported as DDL type, one test is enough.